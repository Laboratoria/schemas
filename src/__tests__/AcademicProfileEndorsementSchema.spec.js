const mongoose = require('mongoose/browser');
const {
  AcademicProfileEndorsementSchema,
  UserSchema,
  AcademicProfileSchema,
  EndorsmentSchema,
} = require('../../')(mongoose);

describe('AcademicProfileEndorsmentSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, AcademicProfileEndorsementSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate example', () => {
    const academicProfile = new mongoose.Document({}, AcademicProfileSchema);
    const endorsement = new mongoose.Document({}, EndorsmentSchema);
    const createBy = new mongoose.Document({}, UserSchema);

    const doc = new mongoose.Document({
      academicProfile: academicProfile._id,
      createBy: createBy._id,
      endorsement: endorsement._id,
      reason: 'foo',
    }, AcademicProfileEndorsementSchema);

    doc.validate();
  });
});
