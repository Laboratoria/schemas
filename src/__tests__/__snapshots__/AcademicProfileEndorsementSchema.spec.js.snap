// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileEndorsmentSchema should fail validation when fields missing 1`] = `
Object {
  "academicProfile": [ValidatorError: Path \`academicProfile\` is required.],
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "endorsement": [ValidatorError: Path \`endorsement\` is required.],
}
`;
